modin/pandas/base.py

```diff
@@ -3242,6 +3242,12 @@ def to_numpy(
         """
         Convert the `BasePandasDataset` to a NumPy array.
         """
+        from modin.config import ExperimentalNumPyAPI
+        if ExperimentalNumPyAPI.get():
+            from ..numpy.arr import array
+
+            return array(_query_compiler=self._query_compiler, _ndim=2)
```

Comment:
I would say this should be put into `modin/pandas/dataframe.py`, not in its base class; `BaseDataset` should contain only common methods for DF and Series, and this piece differs

Reply:
There isn't a `to_numpy` in the DF class, and the current `to_numpy` before this PR differs - do we want to add a `to_numpy` to the DF class? That would make this redundant, so we should probably delete it in that case?

Reply:
Maybe that could be done as a separate refactoring PR, I think there're more methods which we define in Base and override in Series. Let's leave this particular PR in this state for now.